src/mpid/ch4/netmod/ofi/ofi_probe.h

```diff
@@ -129,11 +129,15 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_NM_mpi_improbe(int source,
 
     int vni_src, vni_dst;
     MPIDI_OFI_PROBE_VNIS(vni_src, vni_dst);
-    MPID_THREAD_CS_ENTER(VCI, MPIDI_VCI(vni_dst).lock);
+    if (MPIDI_CH4_MT_MODEL != MPIDI_CH4_MT_LOCKLESS) {
+        MPID_THREAD_CS_ENTER(VCI, MPIDI_VCI(vni_dst).lock);
+    }
```

Comment:
> LOCKLESS essentially is moving the lock from ofi-layer into libfabric layer, right? So I quite doubt the benefit. Do we have convincing measurements to show the benefit? Otherwise, this is just adding complications to the code.

Correct that LOCKLESS moves lock into libfabric and also into the hardware, if it provides the support. So, I think, we'd be able to collect helpful performance data when we get the hardware and do the measurements there.